src/v/cluster/cluster_utils.cc

```diff
@@ -174,13 +174,27 @@ ss::future<> update_broker_client(
 model::broker make_self_broker(const config::node_config& node_cfg) {
     auto kafka_addr = node_cfg.advertised_kafka_api();
     auto rpc_addr = node_cfg.advertised_rpc_api();
+
+    // Calculate memory size
+    const auto shard_mem = ss::memory::stats();
+    uint64_t total_mem = shard_mem.total_memory() * ss::smp::count;
+    uint32_t total_mem_gb = total_mem >> 30;
+
+    // Calculate disk size
+    auto space_info = std::filesystem::space(
+      config::node().data_directory().path);
```

Comment:
What is the context this is called in that makes it ok to call syscalls directly?